File: 798A.py
a = str(input())
n = len(a)
count = 0
for i in range(len(a)//2):
    if(a[i] != a[n-i-1]):
        count += 1
if(count==1 or (count==0 and len(a)%2!=0)):
    print("YES")
else:
    print("NO")

File: 798A.c
#include <stdio.h>
#include <string.h>

int main(){
    char a[100];
    int n , i , count=0;
    scanf("%s",a);
    n = strlen(a);
    for(i=0;i<n/2;i++){
        if(a[i]!=a[n-i-1]){
            count++;
        }
    }
    if(count==1 || (count==0 && n%2!=0)){
        printf("YES");
    }
    else{
        printf("NO");
    }
}